Flaky test AppWithDepsTests.ChangeFileInDependency

```
System.TimeoutException : Task exceeded max running time of 30s at C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\test\Microsoft.DotNet.Watcher.Tools.FunctionalTests\AppWithDepsTests.cs:24
   at System.Threading.Tasks.TaskExtensions.<OrTimeout>d__1.MoveNext() in C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\test\Microsoft.DotNet.Watcher.Tools.Tests\Utilities\TaskExtensions.cs:line 27
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.DotNet.Watcher.Tools.FunctionalTests.AppWithDepsTests.<ChangeFileInDependency>d__2.MoveNext() in C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\test\Microsoft.DotNet.Watcher.Tools.FunctionalTests\AppWithDepsTests.cs:line 24
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

This is a weird regression. Here is the error produced within the "dotnet-build" executed in the test.
```
Process exited 1980
Process exited 1980 with code 0
Building AppWithDeps
Project directory: 'C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps'
Started process 2856: C:\b\w\d6337ac2abf63a4b\.build\dotnet\dotnet.exe build
Microsoft (R) Build Engine version 15.1.548.43366
Copyright (C) Microsoft Corporation. All rights reserved.

  Dependency -> C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\Dependency\bin\Release\netstandard1.5\Dependency.dll
CSC : error CS2012: Cannot open 'C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\obj\Release\netcoreapp1.0\AppWithDeps.dll' for writing -- 'Could not find a part of the path 'C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\obj\Release\netcoreapp1.0\AppWithDeps.dll'.' [C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\AppWithDeps.csproj]
C:\b\w\d6337ac2abf63a4b\.build\dotnet\sdk\1.0.1\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.targets(92,5): error MSB4018: The "GenerateDepsFile" task failed unexpectedly. [C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\AppWithDeps.csproj]
C:\b\w\d6337ac2abf63a4b\.build\dotnet\sdk\1.0.1\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.targets(92,5): error MSB4018: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\bin\Release\netcoreapp1.0\AppWithDeps.deps.json'. [C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\AppWithDeps.csproj]
C:\b\w\d6337ac2abf63a4b\.build\dotnet\sdk\1.0.1\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.targets(92,5): error MSB4018:    at System.IO.Win32FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent) [C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\AppWithDeps.csproj]
C:\b\w\d6337ac2abf63a4b\.build\dotnet\sdk\1.0.1\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.targets(92,5): error MSB4018:    at System.IO.Win32FileSystem.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent) [C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\AppWithDeps.csproj]
C:\b\w\d6337ac2abf63a4b\.build\dotnet\sdk\1.0.1\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.targets(92,5): error MSB4018:    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) [C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\AppWithDeps.csproj]
C:\b\w\d6337ac2abf63a4b\.build\dotnet\sdk\1.0.1\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.targets(92,5): error MSB4018:    at Microsoft.NET.Build.Tasks.GenerateDepsFile.ExecuteCore() [C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\AppWithDeps.csproj]
C:\b\w\d6337ac2abf63a4b\.build\dotnet\sdk\1.0.1\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.targets(92,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() [C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\AppWithDeps.csproj]
C:\b\w\d6337ac2abf63a4b\.build\dotnet\sdk\1.0.1\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.targets(92,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\AppWithDeps.csproj]
C:\b\w\d6337ac2abf63a4b\.build\dotnet\sdk\1.0.1\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.targets(92,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__25.MoveNext() [C:\b\w\d6337ac2abf63a4b\.repositories\DotNetTools\testWorkDir\fb749415b9614031b47f4bf2a742df24\work\AppWithDeps\AppWithDeps.csproj]

Build FAILED.
```

--- a/test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/AppWithDepsTests.cs
+++ b/test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/AppWithDepsTests.cs
@@ -21,13 +21,13 @@ namespace Microsoft.DotNet.Watcher.Tools.FunctionalTests
         [Fact]
         public async Task ChangeFileInDependency()
         {
-            await _app.StartWatcherAsync().OrTimeout();
+            await _app.StartWatcherAsync();
 
             var fileToChange = Path.Combine(_app.DependencyFolder, "Foo.cs");
             var programCs = File.ReadAllText(fileToChange);
             File.WriteAllText(fileToChange, programCs);
 
-            await _app.HasRestarted().OrTimeout();
+            await _app.HasRestarted();
         }
 
         public void Dispose()

--- a/test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/GlobbingAppTests.cs
+++ b/test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/GlobbingAppTests.cs
@@ -24,84 +24,84 @@ namespace Microsoft.DotNet.Watcher.Tools.FunctionalTests
         [InlineData(false)]
         public async Task ChangeCompiledFile(bool usePollingWatcher)
         {
-            await _app.StartWatcherAsync().OrTimeout();
+            await _app.StartWatcherAsync();
 
-            var types = await _app.GetCompiledAppDefinedTypes().OrTimeout();
+            var types = await _app.GetCompiledAppDefinedTypes();
             Assert.Equal(2, types);
 
             var fileToChange = Path.Combine(_app.SourceDirectory, "include", "Foo.cs");
             var programCs = File.ReadAllText(fileToChange);
             File.WriteAllText(fileToChange, programCs);
 
-            await _app.HasRestarted().OrTimeout();
-            types = await _app.GetCompiledAppDefinedTypes().OrTimeout();
+            await _app.HasRestarted();
+            types = await _app.GetCompiledAppDefinedTypes();
             Assert.Equal(2, types);
         }
 
         [Fact(Skip = "Broken. See https://github.com/aspnet/DotNetTools/issues/212")]
         public async Task AddCompiledFile()
         {
-            await _app.StartWatcherAsync().OrTimeout();
+            await _app.StartWatcherAsync();
 
-            var types = await _app.GetCompiledAppDefinedTypes().OrTimeout();
+            var types = await _app.GetCompiledAppDefinedTypes();
             Assert.Equal(2, types);
 
             var fileToChange = Path.Combine(_app.SourceDirectory, "include", "Bar.cs");
             File.WriteAllText(fileToChange, "public class Bar {}");
 
-            await _app.HasRestarted().OrTimeout();
-            types = await _app.GetCompiledAppDefinedTypes().OrTimeout();
+            await _app.HasRestarted();
+            types = await _app.GetCompiledAppDefinedTypes();
             Assert.Equal(3, types);
         }
 
         [Fact]
         public async Task DeleteCompiledFile()
         {
-            await _app.StartWatcherAsync().OrTimeout();
+            await _app.StartWatcherAsync();
 
-            var types = await _app.GetCompiledAppDefinedTypes().OrTimeout();
+            var types = await _app.GetCompiledAppDefinedTypes();
             Assert.Equal(2, types);
 
             var fileToChange = Path.Combine(_app.SourceDirectory, "include", "Foo.cs");
             File.Delete(fileToChange);
 
-            await _app.HasRestarted().OrTimeout();
-            types = await _app.GetCompiledAppDefinedTypes().OrTimeout();
+            await _app.HasRestarted();
+            types = await _app.GetCompiledAppDefinedTypes();
             Assert.Equal(1, types);
         }
 
         [Fact]
         public async Task DeleteSourceFolder()
         {
-            await _app.StartWatcherAsync().OrTimeout();
+            await _app.StartWatcherAsync();
 
-            var types = await _app.GetCompiledAppDefinedTypes().OrTimeout();
+            var types = await _app.GetCompiledAppDefinedTypes();
             Assert.Equal(2, types);
 
             var folderToDelete = Path.Combine(_app.SourceDirectory, "include");
             Directory.Delete(folderToDelete, recursive: true);
 
-            await _app.HasRestarted().OrTimeout();
-            types = await _app.GetCompiledAppDefinedTypes().OrTimeout();
+            await _app.HasRestarted();
+            types = await _app.GetCompiledAppDefinedTypes();
             Assert.Equal(1, types);
         }
 
         [Fact]
         public async Task RenameCompiledFile()
         {
-            await _app.StartWatcherAsync().OrTimeout();
+            await _app.StartWatcherAsync();
 
             var oldFile = Path.Combine(_app.SourceDirectory, "include", "Foo.cs");
             var newFile = Path.Combine(_app.SourceDirectory, "include", "Foo_new.cs");
             File.Move(oldFile, newFile);
 
-            await _app.HasRestarted().OrTimeout();
+            await _app.HasRestarted();
         }
 
         [Fact]
         public async Task ChangeExcludedFile()
         {
-            await _app.StartWatcherAsync().OrTimeout();
+            await _app.StartWatcherAsync();
 
             var changedFile = Path.Combine(_app.SourceDirectory, "exclude", "Baz.cs");
             File.WriteAllText(changedFile, "");
@@ -143,7 +143,7 @@ namespace Microsoft.DotNet.Watcher.Tools.FunctionalTests
 
             public async Task<int> GetCompiledAppDefinedTypes()
             {
-                var definedTypesMessage = await Process.GetOutputLineAsync(m => m.StartsWith("Defined types = "));
+                var definedTypesMessage = await Process.GetOutputLineAsync(m => m.StartsWith("Defined types = ")).OrTimeout();
                 return int.Parse(definedTypesMessage.Split('=').Last());
             }
         }

--- a/test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/NoDepsAppTests.cs
+++ b/test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/NoDepsAppTests.cs
@@ -22,16 +22,16 @@ namespace Microsoft.DotNet.Watcher.Tools.FunctionalTests
         [Fact]
         public async Task RestartProcessOnFileChange()
         {
-            await _app.StartWatcherAsync(new[] { "--no-exit" }).OrTimeout();
-            var pid = await _app.GetProcessId().OrTimeout();
+            await _app.StartWatcherAsync(new[] { "--no-exit" });
+            var pid = await _app.GetProcessId();
 
             // Then wait for it to restart when we change a file
             var fileToChange = Path.Combine(_app.SourceDirectory, "Program.cs");
             var programCs = File.ReadAllText(fileToChange);
             File.WriteAllText(fileToChange, programCs);
 
-            await _app.HasRestarted().OrTimeout();
-            var pid2 = await _app.GetProcessId().OrTimeout();
+            await _app.HasRestarted();
+            var pid2 = await _app.GetProcessId();
             Assert.NotEqual(pid, pid2);
 
             // first app should have shut down
@@ -41,18 +41,18 @@ namespace Microsoft.DotNet.Watcher.Tools.FunctionalTests
         [Fact]
         public async Task RestartProcessThatTerminatesAfterFileChange()
         {
-            await _app.StartWatcherAsync().OrTimeout();
-            var pid = await _app.GetProcessId().OrTimeout();
-            await _app.HasExited().OrTimeout(); // process should exit after run
+            await _app.StartWatcherAsync();
+            var pid = await _app.GetProcessId();
+            await _app.HasExited(); // process should exit after run
 
             var fileToChange = Path.Combine(_app.SourceDirectory, "Program.cs");
             var programCs = File.ReadAllText(fileToChange);
             File.WriteAllText(fileToChange, programCs);
 
-            await _app.HasRestarted().OrTimeout();
-            var pid2 = await _app.GetProcessId().OrTimeout();
+            await _app.HasRestarted();
+            var pid2 = await _app.GetProcessId();
             Assert.NotEqual(pid, pid2);
-            await _app.HasExited().OrTimeout(); // process should exit after run
+            await _app.HasExited(); // process should exit after run
         }
 
         public void Dispose()

--- a/test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/Scenario/WatchableApp.cs
+++ b/test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/Scenario/WatchableApp.cs
@@ -38,16 +38,16 @@ namespace Microsoft.DotNet.Watcher.Tools.FunctionalTests
         public string SourceDirectory { get; }
 
         public Task HasRestarted()
-            => Process.GetOutputLineAsync(StartedMessage);
+            => Process.GetOutputLineAsync(StartedMessage).OrTimeout();
 
         public Task HasExited()
-            => Process.GetOutputLineAsync(ExitingMessage);
+            => Process.GetOutputLineAsync(ExitingMessage).OrTimeout();
 
         public bool UsePollingWatcher { get; set; }
 
         public async Task<int> GetProcessId()
         {
-            var line = await Process.GetOutputLineAsync(l => l.StartsWith("PID ="));
+            var line = await Process.GetOutputLineAsync(l => l.StartsWith("PID =")).OrTimeout();
             var pid = line.Split('=').Last();
             return int.Parse(pid);
         }
@@ -93,7 +93,7 @@ namespace Microsoft.DotNet.Watcher.Tools.FunctionalTests
 
             var args = new[] { "run", "--" }.Concat(arguments);
             Start(args, name);
-            await Process.GetOutputLineAsync(StartedMessage);
+            await Process.GetOutputLineAsync(StartedMessage).OrTimeout();
         }
 
         public virtual void Dispose()

